docs/gitbook/tutorials/prometheus-operator.md

```diff
@@ -0,0 +1,172 @@
+
+Install Prometheus-Operator with Helm v3:
+
+```bash
+helm repo add stable https://kubernetes-charts.storage.googleapis.com
+helm repo update
+kubectl create ns prometheus
+helm upgrade -i prometheus stable/prometheus-operator \
+--namespace prometheus \
+--set prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues=false
+```
+
+The `prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues=false` option allows Prometheus-Operator to watch serviceMonitor outside of his namespace.
+
+You can also set `prometheus.service.type=nodePort` if you want to have
+
+Install Flagger with Helm v3:
+
+```bash
+helm repo add flagger https://flagger.app
+helm repo update
+kubectl create ns flagger
+helm upgrade -i flagger flagger/flagger \
+--namespace flagger \
+--set metricsServer=http://prometheus-prometheus-oper-prometheus.prometheus:9090 \
+--set meshProvider=kubernetes
+```
+
+The `meshProvider` option can be changed to your value, if you want to do something else than Blue/Green Deployment
+
+Install Flagger Loadtester with Helm v3:
+
+```bash
+helm repo add flagger https://flagger.app
+helm repo update
+kubectl create ns flagger
+helm upgrade -i loadtester flagger/loadtester \
+--namespace flagger
+```
+
+Install podinfo with Helm v3:
+
+```bash
+helm repo add sp https://stefanprodan.github.io/podinfo
+helm repo update
+kubectl create ns test
+helm upgrade -i podinfo sp/podinfo \
+--namespace test
+```
+
+## Setting ServiceMonitor
+
+Prometheus Operator is using mostly serviceMonitor instead of annotations.  
+In order to catch metrics for primary and canary service, you will need to create 2 serviceMonitors :
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: podinfo
+  namespace: test
+spec:
+  endpoints:
+  - path: /metrics
+    port: http
+    interval: 15s
+  selector:
+    matchLabels:
+      app: podinfo
+```
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: podinfo-canary
+  namespace: test
+spec:
+  endpoints:
+  - path: /metrics
+    port: http
+    interval: 15s
+  selector:
+    matchLabels:
+      app: podinfo-canary
+```
+
+We are setting `interval: 15s` to have a more aggressive scraping
+If you do not define it, you must to use a longer interval in the Canary object
+
+## Setting Custom metrics
+
+Prometheus Operator is relabeling for every serviceMonitor, you can create custom metrics to you own filter.
+
+```yaml
+apiVersion: flagger.app/v1beta1
+kind: MetricTemplate
+metadata:
+  name: request-success-rate
+  namespace: test
+spec:
+  provider:
+    address: http://prometheus-prometheus-oper-prometheus.prometheus:9090
+    type: prometheus
+  query: rate(http_requests_total{namespace="{{ namespace }}",job="{{ target }}-canary",status!~"5.*"}[{{ interval }}]) / rate(http_requests_total{namespace="{{ namespace }}",job="{{ target }}-canary"}[{{ interval }}]) * 100
```

Comment:
Please format this so it's readable, there are many examples in the docs if you want to see how it should look.